content/guides/references/migration-guide.md

```diff
@@ -2,7 +2,361 @@
+
+```js
+cy.findByTestId('popover').findByRole('button', { expanded: true }).click()
+
+cy.findByTestId('popover')
+  .findByRole('button', { expanded: true }) // A button which matches here (is expanded)...
+  .should('have.attr', 'aria-expanded', 'false') // ...will never pass this assertion.
+```
+
+You can rewrite tests like this to be more specific; in our case, we changed the
+alias to be the first button rather than the unexpanded button.
+
+```js
+cy.findByTestId('popover').findAllByRole('button').first().as('button')
+```
+
+### Command / Cypress API Changes
+
+#### `Cypress.Cookies.defaults` and `Cypress.Cookies.preserveOnce`
+
+The `Cypress.Cookies.defaults` and `CypressCookies.preserveOnce` APIs been
+removed. Use the [`cy.session()`](/api/commands/session) command to preserve
+cookies (and local and session storage) between tests.
+
+```diff
+describe('Dashboard', () => {
+  beforeEach(() => {
+-    cy.login()
+-    Cypress.Cookies.preserveOnce('session_id', 'remember_token')
++    cy.session('unique_identifier', cy.login, {
++       validate () {
++        cy.getCookies().should('have.length', 2)
++       },
++       cacheAcrossSpecs: true
++    })
+  })
+```
+
+#### `cy.server()`, `cy.route()` and `Cypress.Server.defaults`
+
+The` cy.server()` and` cy.route()` commands and the `Cypress.server.defaults`
+API has been removed. Use the [`cy.intercept()`](/api/commands/intercept)
+command instead.
+
+```diff
+  it('can encode + decode headers', () => {
+-   Cypress.Server.defaults({
+-     delay: 500,
+-     method: 'GET',
+-   })
+-   cy.server()
+-   cy.route(/api/, () => {
+-      return {
+-        'test': 'We’ll',
+-      }
+-    }).as('getApi')
++   cy.intercept('GET', /api/, (req) => {
++      req.on('response', (res) => {
++        res.setDelay(500)
++      })
++      req.body.'test': 'We’ll'
++    }).as('getApi')
+    cy.visit('/index.html')
+    cy.window().then((win) => {
+      const xhr = new win.XMLHttpRequest
+      xhr.open('GET', '/api/v1/foo/bar?a=42')
+      xhr.send()
+    })
+
+    cy.wait('@getApi')
+-   .its('url').should('include', 'api/v1')
++   .its('request.url').should('include', 'api/v1')
+  })
+```
+
+#### `.invoke()`
+
+[`.invoke()`](/api/commands/invoke) now throws an error if the function returns
+a promise. If you wish to call a method that returns a promise and wait for it
+to resolve, use [`.then()`](/api/commands/then) instead of `.invoke()`.
+
+```diff
+cy.wrap(myAPI)
+-  .invoke('makeARequest', 'http://example.com')
++  .then(api => api.makeARequest('http://example.com'))
+   .then(res => { ...handle response... })
+```
+
+If `.invoke()` is followed by additional commands or assertions, it will call
+the named function multiple times. This has the benefit that the chained
+assertions can more reliably use the function's return value.
+
+If this behavior is undesirable because you expect the function to invoked only
+once, break the command chain and move the chain's commands and/or assertions to
```

Comment:
```suggestion
once, break the command chain and move the chain'ed commands and/or assertions to
```

Reply:
Don't want the apostrophe, chained not chain'ed

Reply:
🤦🏻‍♀️ whoops. nice catch.